Cabal/Distribution/Simple/PreProcess.hs

```diff
@@ -525,7 +541,7 @@ getCppOptions bi lbi
     = platformDefines lbi
    ++ cppOptions bi
    ++ ["-I" ++ dir | dir <- PD.includeDirs bi]
-   ++ [opt | opt@('-':c:_) <- PD.ccOptions bi, c `elem` "DIU"]
+   ++ [opt | opt@('-':c:_) <- PD.ccOptions bi ++ PD.cxxOptions bi, c `elem` "DIU"]
```

Comment:
I have a vague recollection of something breaking without the `cxx-options` being added here, but that was over a month ago so I don't recall exactly what it was.

Reply:
It does look like it works at the moment for the common case, and without this change C++ CPP options won't be plumbed through at all. I think the ideal solution here would be to have separate `getCppOptionsForHs`, `getCppOptionsForCxx`, `getCppOptionsForC`, etc functions, but, like I said, it's already a bit messy; I think it's sensible to leave that for later.

Reply:
I agree that ideally this would be refactored out into separate functions.

However, I think leaving this for later would be best. I've already invested a lot of _company time_ in this pull request and need to get back to implementing new features on our product.

I added a TODO above this function. Should I add a GitHub issue for this so that it is visibly tracked?